miniengineer/lunch-buddy
# Authentication with devise gem - phase 1

### What this PR does?

- Add devise gem to the project
- Create a custom `RegistrationsController` to handle User registration

### How to test

- start rails server `rails s`
- run `rails db:migrate` to create a `users` table
- open any API testing tool (e.g. [Insomnia](https://insomnia.rest/))
- send a `POST` request to `http://localhost:3000/api/v1/sign_up` with email and password

```
{
	"user": {
		"email": "mini@mail.com",
		"password": "123456"
	}
}
```

- you should get `200 OK` and `User created successfully` message with the user id
- send the request again with the same email and you should get a `404 Bad Request` with `Email has already been taken` message

### Screenshots
<img width="799" alt="Screen Shot 2021-10-17 at 16 43 10" src="https://user-images.githubusercontent.com/47711385/137618080-f19a7e56-48d3-4142-91fc-d193e4795d90.png">

<img width="801" alt="Screen Shot 2021-10-17 at 16 42 59" src="https://user-images.githubusercontent.com/47711385/137618077-b81c7fd4-b5b1-4eb2-afa2-cb7e3a9af490.png">

### Questions or comments
For now the successful registration returns a `user_id` but of course this is a subject to change
